src/main/java/com/bromano/instrumentsgecko/InstrumentsParser.kt

```diff
@@ -384,9 +384,51 @@ object InstrumentsParser {
         // Remove XML Prolog (<xml? ... >) since parser can't handle it
         val trimmedXmlStr = xmlStr.split("\n", limit = 2)[1]
 
-        return DocumentBuilderFactory.newInstance()
-            .newDocumentBuilder()
-            .parse(InputSource(StringReader(trimmedXmlStr)))
+        // Save original system properties
+        val originalProperties = mutableMapOf<String, String?>()
+        val securityProperties = listOf(
+            "jdk.xml.maxGeneralEntitySizeLimit",
+            "jdk.xml.maxParameterEntitySizeLimit", 
+            "jdk.xml.entityExpansionLimit",
+            "jdk.xml.elementAttributeLimit",
+            "jdk.xml.maxXMLNameLimit",
+            "jdk.xml.totalEntitySizeLimit"
+        )
+
+        try {
+            // Set system properties to disable limits
+            securityProperties.forEach { prop ->
+                originalProperties[prop] = System.getProperty(prop)
+                System.setProperty(prop, "0") // 0 means unlimited for most properties
+            }
+
+            val factory = DocumentBuilderFactory.newInstance()
+            // Try to set factory attributes as well (fallback)
+            try {
```

Comment:
I wasn't able to tell from looking online whether this approach actually works. Can you help confirm with a large XML file that DocumentBuilderFactory can be made to support larger XML files through these attributes.

If they can't be, what I'd prefer instead is just having all these system properties set up front once in the Main method of the CLI tool and just having the updated values persist for the entire lifetime of the application. That would avoid needing to keep setting and unsetting the system properties.